### ISSUE TITLE
Started looking at Artwork Beef inside DialogVideoInfo_Includes

### ISSUE THREAD

**Issue by jojobrogess** [SerpentDrago/skin.auramod#94]
https://github.com/SerpentDrago/skin.auramod/blob/9b4b7cd8f3898fd49383a8b5c56e02931e5e6fb9/1080i/DialogVideoInfo-Includes.xml#L836

I couldn't figure out where that code is represented inside the skin.  or what it actually did lol I can somewhat remember having an option for artwork beef within the skin settings itself. but it's been too long. 
Looking at the lines above that cut off, lead me down even further rabbit holes.
these are the two buttons prior. which make no sense to me lol 
neither label can be localized(because they're not in the language files), so I wasn't able to connect it to something within the skin to figure it out. 
Maybe it's left over code? or just wasn't labeled correctly?
```
    <control type="button" id="9006">
      <label>20339</label>
      <onclick>SetFocus(9503)</onclick>
      <visible>String.IsEqual(ListItem.DBType,movie)</visible>
      <visible>!String.IsEmpty(ListItem.IMDBNumber)</visible>
      <include>Defs_Hidden_Button</include>
    </control>
    <control type="button" id="10">
      <description>Artwork (Get Thumb)</description>
      <label>$LOCALIZE[13405]</label>
      <include>Defs_Hidden_Button</include>
    </control>
```

Anyways that's not what I'm asking. 
I was looking at the code for artwork beef/ artwork downloader and was wondering if it was even necessary anymore?
From what I've been able to gather, artwork beef/downloader are both abandoned and depreciated since library scrapers now basically do the same thing. And the "predecessor" is artwork dump, which works completely differently in that it only downloads artwork into your library folders.

Is there some type of behind the scenes idea for this? Or is this something that's just going to have to be removed?

**Comment by inb4after** [comment]
This code refers to old addons and will need to be removed/changed at a later date. The buttons to `Get Artwork` will only be visible if ArtworkDownloader or ArtworkBeef are installed, and since they're not Matrix compatible, the buttons will never be visible. ArtworkDump would be a good replacement for this functionality.

Looks like `button id="10"` is a built-in Kodi button with built-in Kodi localization for its label `Get thumb` ([link](https://github.com/xbmc/xbmc/blob/c3d99abeadfbdff53e18147572ee35e62e81c697/addons/resource.language.en_gb/resources/strings.po#L7004)).

The label for `button id="9006"` is probably `Director` ([link](https://github.com/xbmc/xbmc/blob/c3d99abeadfbdff53e18147572ee35e62e81c697/addons/resource.language.en_gb/resources/strings.po#L12991)) which is grabbed from the default Kodi strings.po, but it's not `$LOCALIZE[]`'d so it won't be translated into other languages. 

Click [here](https://github.com/xbmc/xbmc/blob/master/addons/resource.language.en_gb/resources/strings.po) to view Kodi's strings.po.

**Comment by SerpentDrago** [comment]
yeh old functionality , i'm all for removing it. you are correct the python tmdb library scrapers do all this now

**Comment by wsx79** [comment]
There is an ArtworkDownloader version that work very well for Matrix so I think it's not necessary to delete the refers.
I commit new dialogvideoinfo without artworkbeef (commit #119 

Matrix Artwork Downloader working version
https://github.com/Angelinas1/script.artwork.downloader

**Comment by inb4after** [comment]
> There is an ArtworkDownloader version that work very well for Matrix so I think it's not necessary to delete the refers.
> I commit new dialogvideoinfo without artworkbeef (commit #119 
> 
> Matrix Artwork Downloader working version
> https://github.com/Angelinas1/script.artwork.downloader

Also looks like Artwork Beef was recently ported to py3 as well: https://www.reddit.com/r/Addons4Kodi/comments/ntigl0/artwork_beef_for_kodi_matrix_work_in_progress/

Maybe removing this functionality is not needed?

**Comment by jojobrogess** [comment]
So a while back I did some digging and found this post(the top one): 
https://forum.kodi.tv/showthread.php?tid=258886

In it, rmrector of ArtworkBeef said this:

>It is developed for Kodi 17 Krypton and 18 Leia, and should also work for 16 Jarvis. 15 Isengard may still work, but it's really time to update.
>
>19 Matrix is not supported - AB was a stopgap measure for functionality missing from Kodi ("generally intended to handle extended artwork just as Kodi and scrapers already do for basic artwork"), and now that gap has been closed. For Kodi 19 Matrix, Artwork Dump can be used to download artwork added by scrapers.
 
It also looks like Artwork Downloader has the same type of functionality:
https://kodi.wiki/view/Add-on:Artwork_Downloader
>Downloads all available artwork for TV shows, Movies and Musicvideos in your library.

And Artwork Dump basically moves scraper provided artwork into your library rather than local cache:
>Artwork Dump is an add-on for Kodi that downloads artwork for media in your video and music libraries into your media file system. It only downloads artwork **_already added to your libraries_** by scrapers, NFO files, or other add-ons.

That's why I got the impression that it was a depreciated feature, as well as replacing with Artwork dump would be a little more user specific and out of the scope of the basic functionality of the skin(but could be nice, cuz why not the skin does everything else lol although I can say it is a little buggy for me or was a month ago when I tested it last. maybe wrong settings? idk).  

But anyways that's why I questioned removing them.

Plus at the same time I am a little lost on what the actual functionality of all of them are in the first place in relation to what Kodi already scrapes. From what I can remember of my kodi 18 version, I struggled to scrape all relevant art for videos as well as display them in a timely manner, so I used artwork beef to download them and have them in my media library for speed/inventory purposes. But now kodi 19 gets everything the first try( or at least mostly if the movie isn't too unpopular/old/small budget/has low amount of art on sites anyways) and it and the auramod skin seem to load things with less lag now, at least for me. Can someone shed some light on how this would effect the skin? I can remember seeing somewhere that extra fanart isn't a thing, at least not anymore. 

**AND** I haven't even mentioned Artwork Helper yet. As far as I can tell that's also dead? Or at least hasn't been updated yet?  I can at least understand how the functionality of that would work with the skin. Helps it find stuff lol.  
https://github.com/rmrector/script.artwork.helper
>Artwork Helper provides skins a few options for building a list of images for a multiimage or list control from images available via InfoLabels. Skins can display them without an add-on, but this makes it a bit easier. This is a small add-on that skins can use as a dependency, only doing what a skin asks it to do.
>
>Current version only works for Kodi 17 and 18, see tagged version 0.7.1 for 15 and 16.

This seems to be something that's a little more crucial to the skin. But I don't have it installed, and everything seems good.
As always, I could be wrong lol
